### PR TITLE
avoid creating file autorequire with nil variable

### PR DESCRIPTION
**Pull request by hdeadman**
This is a fix for #166.

### CODE REVIEW

File: lib/puppet/type/archive.rb [hunk 1]
```diff
@@ -160,12 +160,16 @@ def should_to_s(value)
...
     [
       Pathname.new(self[:path]).parent.to_s,
       self[:extract_path],
       '/root/.aws/config',
       '/root/.aws/credentials',
-    ]
+    ].each do |file| 
+      autos << file unless file.nil?
+    end
+    autos
```
Comment (nanliu) [inline lib/puppet/type/archive.rb]: Please just do .compact on the array. Also we might consider only adding /root/.aws on posix platforms.

**Comment by nanliu** [comment]
Would appreciate a spec test supporting this change.

**Comment by hdeadman** [comment]
I updated to use compact. I don't have much Ruby experience but I figured there was a simple way to do it. I am not currently setup to run spec tests, I have only ever written one, but would adding Puppet 4.5 to the travis text matrix cause many of the existing tests to fail (since they don't set the extract_path argument)? The error happens during catalog execution (vs compilation), do tests involve executing the catalog?

**Comment by rnelson0** [comment]
Unit tests (rspec-puppet) are catalog only, testing the state immediately
after compilation. Acceptance tests (beaker) can be used to test the
results of execution.

-- 

Rob Nelson
rnelson0@gmail.com

**Comment by jyaworski** [comment]
Hi @hdeadman. As @rnelson0 mentioned our acceptance testing is a bit sub-par right now. We're actively looking into how to improve this.

That being said, I can accept this PR without a spec test due to its nature. Can you squash your commits?

**Pull request merged**